nmostesting/mocks/Registry.py

```diff
@@ -178,6 +296,31 @@ def check_authorized(self, headers, path, write=False):
 REGISTRY_API = Blueprint('registry_api', __name__)
 
 
+@REGISTRY_API.route('/x-nmos', methods=["GET"], strict_slashes=False)
+def x_nmos():
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+
+    base_data = ['query/', 'registration/']
+
+    return Response(json.dumps(base_data), mimetype='application/json')
+
+
+@REGISTRY_API.route('/x-nmos/registration', methods=["GET"], strict_slashes=False)
+def registration_root():
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    base_data = ['v1.0/', 'v1.1/', 'v1.2/', 'v1.3/']
```

Comment:
This should reflect the enabled version, as we discussed on the call.

Reply:
Hmm, yes we def need to do this for the Query API version (on the `x-nmos/query` endpoints), but I'm not sure about the Registration API version, as that is not something that is being explicitly tested by the current Controller tests.  The Registry now has a `query_api_version` member which can be set as part of tests - perhaps we need a `registration_api_version` member to handle this endpoint (although it won't currently be exercised by the tests)? Alternatively we remove this endpoint as it's not explicitly needed by the current tests (?)

Reply:
Hmm, yes, the mock Registration API is used in the IS-04-01 test suite... I think the [relevant test case](https://github.com/AMWA-TV/nmos-testing/blob/e1edcb9b91e183beb2fa404bc87b43d674aa84f6/nmostesting/suites/IS0401Test.py#L365) checks that the Node registered using the version it was supposed to, which it considers to be the version of the Node API under test, since that is also used to create the DNS-SD advertisement.

I think the same thing is happening here for the DNS-SD advertised Query API version, except that that version comes explicitly from the _Query_ API version spec in the web app for the Controller Tests?

Maybe I've changed my mind then...

* The versions responses of both Query API and Registration API could list all versions from `Config.SPECIFICATIONS["is-04"]["versions"]` (see [here](https://github.com/AMWA-TV/nmos-testing/blob/e1edcb9b91e183beb2fa404bc87b43d674aa84f6/nmostesting/Config.py#L159)).
* The API endpoints below this ought to check the `version` URL parameter matches the API-version-under-test - or store it, like the Registration API ones do, so it can be checked in the test cases?

Reply:
Understood.  I would advocate storing and checking, rather than rejecting, so the test can be explicit about reasons for failure.  Also, I worry that rejecting calls might have unintended consequences for existing test suites.

Reply:
Code changed accordingly.